### PR TITLE
Add react-based frontend

### LINKED ISSUE #68: End to end tests with Playwright
Currently we use some JupyterHub fixtures and `pytest` for testing.

There is also https://github.com/plasmabio/tljh-repo2docker/pull/65 which uses the `pytest-jupyterhub` plugin to streamline testing even more.

It could be interesting to also do more end to end testing, like in the JupyterHub repo:

- https://github.com/jupyterhub/jupyterhub/blob/73b2307b367f8582e015fb6f8da5dd213523a943/jupyterhub/tests/browser/conftest.py#L5-L14
- https://github.com/jupyterhub/jupyterhub/blob/main/jupyterhub/tests/browser/test_browser.py

### PR DESCRIPTION
**Pull request by trungleduc**
## Code changes
- Remove old jinja-based UI
- Add new UI based on React
- Update CI, add isolated and UI tests, close #68  

ci

## Screenshot
hub

- Environment list

hub1

- Environment dialog

hub2

- Server list

hub3

- Server dialog

hub4

- Machine profiles

machine-profile

### CODE REVIEW

**Comment by jtpio** [comment]
Thanks @trungleduc for working on this!

Do you have a screenshot around for what the new frontend / UI looks like, even if it's still a draft for now and subject to changes? Thanks!

**Comment by trungleduc** [comment]
@jtpio I updated the first post.

**Comment by trungleduc** [comment]
@jtpio  could you take a look at this PR? Thanks

**Comment by jtpio** [comment]
Thanks @trungleduc for working on this, this is looking really good!

File: MANIFEST.in [hunk 1]
```diff
@@ -1,3 +1,8 @@
 recursive-include tljh_repo2docker/templates/ *
 recursive-include tljh_repo2docker/static/ *
 recursive-include tljh_repo2docker/tests/ *
+recursive-include frontend/ *
```
Comment (jtpio) [inline MANIFEST.in:4]: Maybe it could also be useful to switch to `hatch` and drop this `MANIFEST.in`? (could be done / tracked separately)

Comment (trungleduc) [reply MANIFEST.in:4]: https://github.com/plasmabio/tljh-repo2docker/issues/71

**Comment by trungleduc** [comment]
@jtpio this PR is ready again

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+jobs:
+  build:
+    name: Build tljh_repo2docker
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Install Conda environment with Micromamba
```
Comment (jtpio) [inline .github/workflows/build.yml:18]: Do we need micromamba, or would `jupyterlab/maintainer-tools/.github/actions/base-setup@v1` be enough?

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+        run: echo "::set-output name=dir::$(yarn cache dir)"
+
+      - name: Setup yarn cache
+        uses: actions/cache@v2
+        id: yarn-cache
+        with:
+          path: ${{ steps.yarn-cache-dir-path.outputs.dir }}
+          key: yarn-${{ hashFiles('./yarn.lock') }}
+          restore-keys: |
+            yarn-
```
Comment (jtpio) [inline .github/workflows/build.yml:49]: These would be handled by `jupyterlab/maintainer-tools/.github/actions/base-setup@v1` (mentioned above).

File: .yarnrc.yml [hunk 4]
```diff
@@ -0,0 +1,2 @@
+nodeLinker: node-modules
```
Comment (jtpio) [inline .yarnrc.yml:1]: If this simplifies the setup, we could also use `npm` instead of `yarn`?

File: tljh_repo2docker/tests/test_images.py [hunk 5]
```diff
@@ -10,7 +10,8 @@ async def test_images_list_admin(app):
     cookies = await app.login_user('admin')
     r = await get_page('environments', app, cookies=cookies, allow_redirects=False)
     r.raise_for_status()
-    assert 'Repository' in r.text
+    print(r.text)
```
Comment (jtpio) [inline tljh_repo2docker/tests/test_images.py:13]: To remove?

```suggestion
```

File: .yarnrc.yml (hunk 4 above)
Comment (trungleduc) [reply .yarnrc.yml]: indeed, updated!

File: .github/workflows/build.yml [hunk 6]
```diff
@@ -0,0 +1,163 @@
...
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Base Setup
+        uses: jupyterlab/maintainer-tools/.github/actions/base-setup@v1
+
+      - name: Lint the application
+        shell: bash -l {0}
```
Comment (jtpio) [inline .github/workflows/build.yml:22]: Maybe this `shell` (and the others below) could be removed if we use the `base-setup` action?

**Comment by jtpio** [comment]
I didn't go through the whole frontend code in the details, but overall it looked good. We can always iterate if later if something needs to be fixed.

**Pull request merged**